### PR TITLE
Glue housenumbers tokens before compiling data

### LINKED ISSUE #286: Dix + huit vs. dix-huit during housenumbers indexation
`dix` and ` huit` are indexed twice when there are both 8, 10 and 18.

```
"SADD" "p|dix" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|dix" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|katorz" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|katr" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|neuf" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|seiz" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|sep" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|sink" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|six" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|troi" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|ui" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
"SADD" "p|ui" "le" "pilori" "aut" "76329" "sein" "valas" "du" "76" "normandi" "gruchet" "maritim" "76210" "ru"
```

### PR DESCRIPTION
**Pull request by yohanboniface**
fix #286

See also https://github.com/addok/addok-france/pull/3

### CODE REVIEW

File: addok/helpers/index.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def prepare_housenumbers(doc):
     if housenumbers:
         doc['housenumbers'] = {}
         for number, data in housenumbers.items():
-            for hn in preprocess(number):
-                data['raw'] = number
-                doc['housenumbers'][str(hn)] = data.copy()
+            token = ''.join(list(preprocess(number)))
+            data['raw'] = number
+            doc['housenumbers'][token] = data.copy()
```
Comment (davidbgk) [inline addok/helpers/index.py]: We don't need a `copy()` anymore.

File: addok/helpers/index.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def prepare_housenumbers(doc):
     if housenumbers:
         doc['housenumbers'] = {}
         for number, data in housenumbers.items():
-            for hn in preprocess(number):
-                data['raw'] = number
-                doc['housenumbers'][str(hn)] = data.copy()
+            token = ''.join(list(preprocess(number)))
```
Comment (davidbgk) [inline addok/helpers/index.py]: We should document why we have to do that.

File: addok/helpers/text.py [hunk 3]
```diff
@@ -56,6 +63,18 @@ def key(self):
...
+        return self._positions
+
+    def _set_position(self, position):
+        if isinstance(position, list):
+            self._positions = position
+        else:
+            self._positions.append(position)
+
+    # Allow to store multiple positions when a token is subdivided.
+    position = property(_get_position, _set_position)
```
Comment (davidbgk) [inline addok/helpers/text.py]: why not `@property` and `@position.setter`?

Comment (yohanboniface) [inline addok/helpers/text.py]: Changed.

**Pull request merged**